server/apiCalls/ugRestApi.js

```diff
@@ -0,0 +1,150 @@
+  const today = new Date()
+  const date = today.getFullYear() + '-' + (today.getMonth() + 1) + '-' + today.getDate()
+  const time = today.getHours() + ':' + today.getMinutes() + ':' + today.getSeconds()
+  const dateTime = date + ' ' + time
+  return dateTime
+}
+
+const generateEmployeeObjectFromGroups = (groupsAlongWithMembers, assistants, teachers, examiners, responsibles) => {
+  // now need to filter groups according to above mentioned groups
+  const employee = {}
+  if (assistants.length) {
+    const assistantGroup = groupsAlongWithMembers.find(x => x.name === assistants[0])
+    if (assistantGroup) {
+      const flatArrWithHtmlStr = createPersonHtml(assistantGroup.members)
+      employee.assistants = flatArrWithHtmlStr
+    }
+  }
+  if (teachers.length) {
+    const teachersGroup = groupsAlongWithMembers.find(x => x.name === teachers[0])
+    if (teachersGroup) {
+      const flatArrWithHtmlStr = createPersonHtml(teachersGroup.members)
+      employee.teachers = flatArrWithHtmlStr
+    }
+  }
+  if (examiners.length) {
+    const examinersGroup = groupsAlongWithMembers.find(x => x.name === examiners[0])
+    if (examinersGroup) {
+      const flatArrWithHtmlStr = createPersonHtml(examinersGroup.members)
+      employee.examiners = flatArrWithHtmlStr
+    }
+  }
+  if (responsibles.length) {
+    const responsiblesGroup = groupsAlongWithMembers.find(x => x.name === responsibles[0])
+    if (responsiblesGroup) {
+      const flatArrWithHtmlStr = createPersonHtml(responsiblesGroup.members)
+      employee.responsibles = flatArrWithHtmlStr
+    }
+  }
+  return employee
+}
+
+/**
+ * This will first prepare filter query then pass that query to ug rest api and then return groups data along with members.
+ * @param assistants Assistants group name
+ * @param teachers Teachers group name
+ * @param examiners Examiners group name
+ * @param responsibles Responsibles group name
+ * @returns Will return groups along with members from ug rest api.
+ */
+async function getAllGroupsAlongWithMembersRelatedToCourse(assistants, teachers, examiners, responsibles) {
+  const { url, key } = serverConfig.ugRestApiURL
+  const { authTokenURL, authClientId, authClientSecret } = serverConfig.ugAuth
+  const ugConnectionProperties = {
+    ugTokenURL: authTokenURL,
+    clientId: authClientId,
+    clientSecret: authClientSecret,
+    ugURL: url,
+    subscriptionKey: key,
+  }
+  ugRestApiHelper.initConnectionProperties(ugConnectionProperties)
+  const filterData = []
+  if (assistants.length) {
+    filterData.push(assistants[0])
+  }
+  if (teachers.length) {
+    filterData.push(teachers[0])
+  }
+  if (examiners.length) {
+    filterData.push(examiners[0])
+  }
+  if (responsibles.length) {
+    filterData.push(responsibles[0])
+  }
+  log.info('Started fetching data of groups along with members from UG Rest Api at ', getCurrentDateTime())
+  log.info('Fetching groups along with members')
+  log.info('Using Filter Query Data', { filterData })
```

Comment:
What is included in filterData? if it is user data, we don't want to push to graycloud log data, because it's personal data, like email (GDPR rules). it is better to print the names of groups and then how many users were fetched,
In old code f.e., printed how many users fetched for courseCode and semester

```
log.info(
      ' Ug Redis fetched correctly,  number of teachers: ',
      arrWithStringifiedArrays[0].length,
      ' for course ',
      courseCode,
      ' for semester ',
      semester
    )
```

Reply:
Added response information.